packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx

```diff
@@ -89,11 +90,21 @@ export class UnthemedTimeRangePicker extends PureComponent<TimeRangePickerProps,
 
     return (
       <ButtonGroup className={styles.container}>
-        {hasAbsolute && <ToolbarButton variant={variant} onClick={onMoveBackward} icon="angle-left" narrow />}
+        {hasAbsolute && (
+          <ToolbarButton
+            aria-label="Move time range backwards"
+            variant={variant}
+            onClick={onMoveBackward}
+            icon="angle-left"
+            narrow
+          />
+        )}
 
         <Tooltip content={<TimePickerTooltip timeRange={value} timeZone={timeZone} />} placement="bottom">
           <ToolbarButton
-            aria-label="TimePicker Open Button"
+            data-testid={selectors.components.TimePicker.openButton}
+            aria-label={`Time range editor ${formattedRange(value, timeZone)}`}
```

Comment:
This works great! 🎉👍

Reply:
I feel we need to use a lot more descriptive text in the Time range editor label text. Currently, the screen reader reads it as "Time range editor {any time range selected}" which might be a bit difficult for the user to comprehend. 

Rather, we could use something like `"Time range editor with current time range {any time range selected}"`